src/Symfony/Component/Console/Output/ConsoleSectionOutput.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Output;
+
+use Symfony\Component\Console\Formatter\OutputFormatterInterface;
+use Symfony\Component\Console\Helper\Helper;
+use Symfony\Component\Console\Terminal;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+class ConsoleSectionOutput extends StreamOutput
+{
+    public $content;
+    public $lines = 0;
+
+    private $sectionReference;
+    private $terminal;
+
+    public function __construct($stream, OutputSectionReference $sectionReference, $verbosity, $decorated, OutputFormatterInterface $formatter)
+    {
+        parent::__construct($stream, $verbosity, $decorated, $formatter);
+        $this->sectionReference = $sectionReference;
+        $this->sectionReference->addRef($this);
+        $this->terminal = new Terminal();
+    }
+
+    /**
+     * Clears previous output for this section.
+     *
+     * @param int $lines Number of lines to clear. If null, then the entire output of this section is cleared
+     */
+    public function clear($lines = null)
+    {
+        if (empty($this->content) || !$this->isDecorated()) {
+            return;
+        }
+
+        $number = $lines ?: $this->lines;
+
+        $buffer = $this->sectionReference->calculateBuffer($this);
+        $buffer += $number;
+
+        $this->content = '';
+        $this->lines = $this->lines - $number;
+
+        if ($buffer > 0) {
+            $this->moveUp($buffer);
+            $this->clearOutput();
+        }
+
+        $this->printOutput();
+    }
+
+    /**
+     * Overwrites the previous output with a new message.
+     *
+     * @param string $message
+     */
+    public function overwrite($message)
+    {
+        $this->clear();
+        $this->writeln($message, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doWrite($message, $newline)
+    {
+        if (!$this->isDecorated()) {
+            return parent::doWrite($message, $newline);
+        }
+
+        $newline = true;
+        $buffer = $this->sectionReference->calculateBuffer($this);
+
+        if ($buffer > 0) {
+            $this->moveUp($buffer);
+            $this->clearOutput();
+        }
+
+        $this->calculateLines($message, $newline);
+
+        parent::doWrite($message, $newline);
+
+        $this->printOutput();
+    }
+
+    private function calculateLines($messages, $newLine)
+    {
+        $total = 0;
+
+        $prevLines = explode("\n", $messages);
+
+        for ($i = 0; $i < count($prevLines); ++$i) {
+            $total += ceil(Helper::strlen($prevLines[$i]) / $this->terminal->getWidth()) ?: 1;
```

Comment:
I don't think it's stille the good way to calculate the length of a line in a console, as an example a tabulation must be count as 8 when displaying (or other number depending on the term configuration) or a colore string / background will have special chars which here will be counted, but they must not.

I was confronted with the same behavior and use this implementation : https://github.com/jolicode/asynit/blob/master/src/Output/TestOutput.php#L110 far from being perfect but gives a good start.